main.cc

```diff
@@ -1085,8 +1085,37 @@ int main(int ac, char** av) {
                 static sharded<alternator::executor> alternator_executor;
                 alternator_executor.start(std::ref(proxy), std::ref(mm)).get();
                 static alternator::server alternator_server(alternator_executor);
-                alternator_server.init(addr, cfg->alternator_port()).get();
-                startlog.info("Alternator server listening on {} port {}", addr, cfg->alternator_port());
+                std::optional<uint16_t> alternator_port;
+                if (cfg->alternator_port()) {
+                    alternator_port = cfg->alternator_port();
+                }
+                std::optional<uint16_t> alternator_tls_port;
+                std::optional<tls::credentials_builder> creds;
+                if (cfg->alternator_tls_port()) {
+                    creds.emplace();
+                    alternator_tls_port = cfg->alternator_tls_port();
+                    creds->set_dh_level(tls::dh_params::level::MEDIUM);
+                    creds->set_x509_key_file(cert, key, tls::x509_crt_format::PEM).get();
+                    if (trust_store.empty()) {
+                        creds->set_system_trust().get();
+                    } else {
+                        creds->set_x509_trust_file(trust_store, tls::x509_crt_format::PEM).get();
+                    }
+                    creds->set_priority_string(db::config::default_tls_priority);
+                    if (!prio.empty()) {
+                        creds->set_priority_string(prio);
+                    }
+                    if (clauth) {
+                        creds->set_client_auth(seastar::tls::client_auth::REQUIRE);
+                    }
+                }
+                alternator_server.init(addr, alternator_port, alternator_tls_port, creds).get();
+                if (alternator_port) {
+                    startlog.info("Alternator server listening on {} port {}", addr, *alternator_port);
```

Comment:
You already printed these messages in init(). So now they appear twice?
Also, since init() doesn't throw if the initialization failed (see my comment above), I think here you print these success messages even if the initialization actually failed?

Reply:
Yes, I'll fix that and test by trying to run a server on a busy port.